### ISSUE TITLE
only warnings,no error,but the training is still 0%

### ISSUE THREAD

**Issue by Jessicall** [AliaksandrSiarohin/first-order-model#91]
(animation1) root@6ffc0be116fa:/remote-home/my/pycharmprojects/first-order-model# CUDA_VISIBLE_DEVICES=0,1 python run.py --config config/fashion-256.yaml --device_ids 0,1

/usr/local/miniconda3/envs/animation1/lib/python3.6/site-packages/dask/config.py:129: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
/usr/local/miniconda3/envs/animation1/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
run.py:40: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
Use predefined train-test split.
Training...
  0%|                                                                                                                                                          | 0/100 [00:00<?, ?it/s]/usr/local/miniconda3/envs/animation1/lib/python3.6/site-packages/torch/nn/functional.py:2423: UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
/usr/local/miniconda3/envs/animation1/lib/python3.6/site-packages/torch/nn/functional.py:1332: UserWarning: nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.
  warnings.warn("nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.")
.......
the code is running but the training is 0% all the time,what's the possible reasons?

**Comment by AliaksandrSiarohin** [comment]
How much time you wait? What is gpu utilisation? Decrease num_repeats in config.

**Comment by Jessicall** [comment]
I wait for 20 minutes and I use two 2080 ti for training
num-epoch is 100
num-repeats is 50
batch size is 27

**Comment by AliaksandrSiarohin** [comment]
Gpu-utilisation? Batch size is too large probably. Reduce to 16

**Comment by ssnake** [comment]
My training's been lasting for almost 2 days but still there is no any progress. It stuck on 0/100. I suspected it might have been because of training data ~4GB...
GPU utilization
```
|   1  GeForce RTX 208...  Off  | 00000000:02:00.0 Off |                  N/A |
| 49%   84C    P2   246W / 250W |   9931MiB / 11019MiB |     99%      Default |

```

**Comment by Honlan** [comment]
I also met the problem .....

**Comment by ssnake** [comment]
Make png files out of each video and put them into separate folders under test or training folders. This helped me

**Comment by Honlan** [comment]
> Make png files out of each video and put them into separate folders under test or training folders. This helped me

Currently, my data looks like this:
```
vox-png
- train
  - Zbz058o9j6s#003945#004076.mp4
    - png
- test
```
I am not quite clear about your answer. Could you please explain more? thanks a lot.

**Comment by ssnake** [comment]
first-order-model
  - config
    - zel-256.yaml
  - data
    - zel
      - video1
        - 1.png
        - 2.png
        ....
        - 43343.png
       - video2
         - 1.png
         ...
         - 344433.png

**Comment by Honlan** [comment]
> first-order-model
> 
> * config
>   
>   * zel-256.yaml
> * data
>   
>   * zel
>     
>     * video1
>       
>       * 1.png
>       * 2.png
>         ....
>       * 43343.png
>     * video2
>       
>       * 1.png
>         ...
>       * 344433.png

where are the `train` and `test` folders? and why a single video contains so many frames?

**Comment by Honlan** [comment]
@AliaksandrSiarohin I found several lines of codes that should be optimized.

In lines `101-103` of `frames_dataset.py`:
```
 if self.is_train and self.id_sampling:
    name = self.videos[idx]
    path = np.random.choice(glob.glob(os.path.join(self.root_dir, name + '*.mp4')))
```
which means that a `glob` operation is conducted for every sample. This may consume much time when there are a lot of `*.mp4` files, especially on a distributed file system.

Instead, it is better to obtain all the video names and store their corresponding `*.mp4` in a dictionary, which can be finished in `__init__` of `FramesDataset` and only requires a single operation of `glob`.

I modified as above and the training process starts to run :)

Thanks again for you awesome work!

**Comment by AliaksandrSiarohin** [comment]
For me glob was not a problem, if a this a problem than yes, the solution with duct is valid. Alternatively lru_cache can be used. 

For vox-celeb there was a naming error, see #137 
Actual one epoch was more iterations that I used.

**Comment by kenmbkr** [comment]
Try printing the losses after this [line](https://github.com/AliaksandrSiarohin/first-order-model/blob/ebb4b9371c585cf6232ded511ebf46aba15e3c2c/train.py#L76) to check if your model is indeed training or stuck somewhere.

For me, multiple GPUs is the problem. I use up to only 80-90% of my GPU memory but training often pauses after the first epoch is done. If I kill the process when training stucks, I can see a thread lock message like [this](https://discuss.pytorch.org/t/trouble-with-multiple-gpu-setup-thread-lock/42791).

This problem did not happen if I resolve to single GPU training.
